pkg/codegen/go/gen_program.go

```diff
@@ -944,7 +944,12 @@ func (g *generator) addPulumiImport(pkg, versionPath, mod, name string) {
 	// module named IndexToken.
 	info, hasInfo := g.getGoPackageInfo(pkg) // We're allowing `info` to be zero-initialized
 	importPath := func(mod string) string {
-		importBasePath := fmt.Sprintf("github.com/pulumi/pulumi-%s/sdk%s/go/%s", pkg, versionPath, pkg)
+		var importBasePath string
+		if descriptor, ok := g.program.PackageDescriptors()[pkg]; ok && descriptor.Name == "terraform-provider" {
+			importBasePath = fmt.Sprintf("github.com/pulumi/pulumi-terraform-provider/sdks/go/%s/v3", pkg)
+		} else {
+			importBasePath = fmt.Sprintf("github.com/pulumi/pulumi-%s/sdk%s/go/%s", pkg, versionPath, pkg)
```

Comment:
Assuming that this is the nub of the change, I'm not sure this is what we want. What I _think_ happens now:

* `pulumi convert` spies a TF provider
* `pulumi convert` generates a PCL program with a `package { ... }` block in it that references the `terraform-provider` provider with an appropriate parameter
* `pulumi convert`, now armed with `package` blocks, can call SDKgen using the parameterized `terraform-provider`
* SDKgen produces an SDK, `@pulumi/tf-foo` say, that registers resources using the `terraform-provider` and the appropriate parameter
* Importantly, the SDK *name* is still `@pulumi/<package-name>`; it's just that the generated SDK implementation will know to use the `terraform-provider` provider.
* Programgen (this bit) *does* need to do some stuff if it's a local SDK, but I don't think this includes changing the name. It's more that e.g. the import path might be `../path/to/local/@pulumi/<package-name>` instead of `@pulumi/<package-name>`, if that makes sense.